todo_app/test/integration_test.py

```diff
@@ -1,51 +1,73 @@
 from todo_app import app
-from todo_app.test.test_data.testing_data import TestData
+import mongomock
+from todo_app.assets.constants import in_progress_status
+from todo_app.data.mongo_db_service import MongoDbService
 
 @pytest.fixture
 def client():
     file_path = find_dotenv('.env.test')
     load_dotenv(file_path, override=True)
+    
+    with mongomock.patch(servers=(('fakemongo.com', 27017),)):
+        test_app = app.create_app()
+        with test_app.test_client() as client:
+            yield client
+            
+def test_index_page(client):
+    # When
+    response = client.get('/')
 
-    test_app = app.create_app()
-
-    with test_app.test_client() as client:
-        yield client
-
-class StubResponse():
-    def __init__(self, fake_response_data):
-        self.fake_response_data = fake_response_data
-
-    def json(self):
-        return self.fake_response_data
+    # Then
+    assert response.status_code == 200
+    print(response.data.decode())
+    assert "Clare's todo list" in response.data.decode()
+    assert 'What do you need to do?' in response.data.decode()
+    assert 'Add item' in response.data.decode()
 
+def test_adding_item(client):
+    # When
+    response = client.post('/add_item', data={
+        'todo': 'Test item',
+    }, follow_redirects=True)
 
-def stub(url, params={}):
-    test_board_id = os.environ.get('TRELLO_BOARD_ID')
+    # Then
+    assert "Clare's todo list" in response.data.decode()
+    assert 'Test item' in response.data.decode()
 
-    url_mappings = {
-        f'https://api.trello.com/1/boards/{test_board_id}/lists': TestData.fake_lists,
-        f'https://api.trello.com/1/boards/{test_board_id}/cards': TestData.fake_cards,
-        f'https://api.trello.com/1/list/{TestData.fake_to_do_list_id}': TestData.fake_to_do_list,
-        f'https://api.trello.com/1/list/{TestData.fake_in_progress_list_id}': TestData.fake_in_progress_list,
-        f'https://api.trello.com/1/list/{TestData.fake_done_list_id}': TestData.fake_done_list
-    }
 
-    fake_response_data = url_mappings.get(url)
-    if fake_response_data is not None:
-        return StubResponse(fake_response_data)
+def test_updating_item_status(client):
+    # Given
+    response = client.post('/add_item', data={
+        'todo': 'Test item',
+    }, follow_redirects=True)
+    itemId = str(MongoDbService().get_items()[0].id)
 
-    raise Exception(f'Integration test did not expect URL "{url}"')
+    # When
+    response = client.post('/update_status', 
+                        json={'itemId': itemId, 'newStatus': in_progress_status}, 
+                        follow_redirects=True)
+    
+    # Then
+    # Verify 'Test item' is in the 'In Progress' column
+    pattern = r'(In Progress)(.|\n)*(Test item)(.|\n)*(Done)'
+    match = re.search(pattern, response.data.decode())
+    assert match is not None
 
-def test_index_page(monkeypatch, client):
-    monkeypatch.setattr(requests, 'get', stub)
+def test_deleting(client):

```

Comment:
Nice, solid extra additions to these integration tests too 😄

Reply:
I meant to leave a comment in the PR about the following by the way:

I couldn't get `with mongomock.patch(servers=(('fakemongo.com', 27017),)):` working for ages and it was because my code was using `from pymongo import MongoClient` instad of `import pymongo` and then using `pymongo.MongoClient` (source of realising this was my issue: https://github.com/mongomock/mongomock)

Which in hindsight does check out with this line in the exercise instructions:

> Your integration test **can now use pymongo.MongoClient** to connect and insert dummy data as if it were interacting with a real database. Delete all of the code related to mocking requests.get.

But wondering if it might be worth an explicit hint/comment in the instructions?

Reply:
Nice spot - it's one I've been a bit lazy about adding because the only people who ever hit it are at SW; nobody else ever re-writes the import statements that we give in examples (and that is definitely a compliment! 😆) but sounds like I should just do it! Thanks for flagging 😄